src/main/java/fr/pickaria/pterodactylpoweraction/ConnectionListener.java

```diff
@@ -60,6 +61,9 @@ public void onServerPreConnect(ServerPreConnectEvent event) {
 
         shutdownManager.cancelTask(originalServer);
 
+        String serverName = originalServer.getServerInfo().getName();
+        boolean isManagedServer = configurationLoader.getConfiguration().getAllServers().contains(serverName);
```

Comment:
This line is repeated at least twice in this file. You can make a new private method to reduce the duplicated code:

```java
private boolean isManagedServer(RegisteredServer server) {
    String serverName = originalServer.getServerInfo().getName();
    return configurationLoader.getConfiguration().getAllServers().contains(serverName);
}
```

```suggestion
        boolean isManagedServer = this.isManagedServer(originalServer);
```